pkg/controller/resourceclaim/controller.go

```diff
@@ -403,6 +496,49 @@ func (ec *Controller) syncPod(ctx context.Context, namespace, name string) error
 		}
 	}
 
+	if pod.Spec.NodeName == "" {
```

Comment:
Question: How is this different to `podNeedsWork`? Seems as if much of the same logic is implemented (although I haven't compared the code paths in depth).

Reply:
The logic in both is very similar. The difference is that `podNeedsWork` gets invoked as part of an informer callback and has to make decisions more quickly and without blocking. Here we do the actual work.